docs/development/kubernetes-clients.md

```diff
@@ -0,0 +1,373 @@
+- JSON merge patches always replace lists as a whole and don't merge them. Keep this in mind when operating on lists with patch requests. If the controller is the exclusive owner of the entire list, it's safe to run without optimistic locking. Though, if you want to prevent overwriting concurrent changes to the list or its items made by other actors (e.g., additions/removals to the `metadata.finalizers` list), enable optimistic locking.
+- Strategic merge patches are able to make more granular modifications to lists and their elements without replacing the entire list. It uses Golang struct tags of the API types to determine which and how lists should be merged. See [this document](https://kubernetes.io/docs/tasks/manage-kubernetes-objects/update-api-object-kubectl-patch/) or the [strategic merge patch documentation](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md) for more in-depth explanations and comparison with JSON merge patches.
+  With this, controllers *might* be able to issue patch requests for individual list items without optimistic locking, even if they are not exclusive owners of the entire list. Remember to check the `patchStrategy` and `patchMergeKey` struct tags of the fields you want to modify before blindly adding patch requests without optimistic locking.
+- Strategic merge patches are only supported by built-in Kubernetes resources and custom resources served by Extension API servers. Strategic merge patches are not supported by custom resources defined by `CustomResourceDefinition`s (see [this comparison](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#advanced-features-and-flexibility)). In that case, fallback to JSON merge patches.
+- [Server-side Apply](https://kubernetes.io/docs/reference/using-api/server-side-apply/) is yet another mechanism to modify API objects, which is supported by all API resources (in newer Kubernetes versions). However, it has a few problems and more caveats preventing us from using it in Gardener at the time of writing. See [gardener/gardener#4122](https://github.com/gardener/gardener/issues/4122) for more details.
+
+> Generally speaking, patches are often the better option compared to update requests because they can save network traffic, encoding/decoding effort and avoid conflicts under the presented conditions.
+> If choosing a patch type, consider which type is supported by the resource you're modifying and how you can avoid conflicts if your modification is safe to run without optimistic locking.
```

Comment:
```suggestion
> If choosing a patch type, consider which type is supported by the resource you're modifying and what will happen in case of a conflict. Consider whether your modification is safe to run without optimistic locking.
```